src/views/InteractiveMap/Escher.vue

```diff
@@ -349,6 +360,37 @@ export default Vue.extend({
       });
       return fluxesFiltered;
     },
+    ecModelFluxes(fluxes) {
+      // Map fluxes from an enzyme-constrained model to the corresponding
+      // escher map, as reaction ids from ecModels have additional prefixes
+      // and/or suffixes.
+      const fluxesMapped = {};
+      Object.keys(fluxes).forEach(rxn => {
+        var newRxn = rxn;
+        var newFlux = fluxes[rxn];
+        if (rxn.startsWith("arm_")) {
+          // For isozymes, use the flux in the reaction "arm_XXX".
+          newRxn = rxn.substring(4);
+        } else if (rxn.endsWith("No1")) {
+          // For single enzymes, use the flux in the reaction "XXXNo1" (for
+          // this, check first that no arm reaction is already present).
+          const rootRxn = rxn.substring(0, rxn.length - 3);
+          if (!Object.keys(fluxesMapped).includes(rootRxn)) {
+            newRxn = rootRxn;
+          }
+        }
+        fluxesMapped[newRxn] = newFlux;
+        if (newRxn.endsWith("_REV") && fluxes[rxn] > 0) {
+          // For reversible enzymes ("XXX_REV") that are active in the backwards
+          // direction, replace the existing flux in the forward reaction by the
+          // negative value.
+          newRxn = newRxn.substring(0, newRxn.length - 4);
+          newFlux = newFlux * -1;
+          fluxesMapped[newRxn] = newFlux;
```

Comment:
I think `Object.keys()` order is not deterministic, so could this be overwritten by the forwards direction with the same name if it comes after the reverse?

Reply:
fixed in [`3027223`](https://github.com/DD-DeCaF/caffeine-vue/pull/171/commits/30272236089b7fdba081fff39e5ccf59f2f2ff70): by doing the replacement of isozymes and reversibilities in 2 separate `forEach` blocks the order in which the reactions come will not matter. Also, by matching `arm_XXX` in the old dict instead of `XXX` in the new one ([line 377](https://github.com/DD-DeCaF/caffeine-vue/pull/171/files#diff-cb5e31d152203986c650570c10bfa708R377)), the order of isozymes and single enzymes will also not matter.